mteb/tasks/Classification/pol/PolHateClassification.py

```diff
@@ -0,0 +1,83 @@
+        socioeconomic_status="high",
+        annotations_creators="expert-annotated",
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="""
+        @inproceedings{rottger-etal-2022-multilingual,
+            title = "Multilingual {H}ate{C}heck: Functional Tests for Multilingual Hate Speech Detection Models",
+            author = {R{\"o}ttger, Paul  and
+            Seelawi, Haitham  and
+            Nozza, Debora  and
+            Talat, Zeerak  and
+            Vidgen, Bertie},
+            editor = "Narang, Kanika  and
+            Mostafazadeh Davani, Aida  and
+            Mathias, Lambert  and
+            Vidgen, Bertie  and
+            Talat, Zeerak",
+            booktitle = "Proceedings of the Sixth Workshop on Online Abuse and Harms (WOAH)",
+            month = jul,
+            year = "2022",
+            address = "Seattle, Washington (Hybrid)",
+            publisher = "Association for Computational Linguistics",
+            url = "https://aclanthology.org/2022.woah-1.15",
+            doi = "10.18653/v1/2022.woah-1.15",
+            pages = "154--169",
+            abstract = "Hate speech detection models are typically evaluated on held-out test sets. However, this risks painting an incomplete and potentially misleading picture of model performance because of increasingly well-documented systematic gaps and biases in hate speech datasets. To enable more targeted diagnostic insights, recent research has thus introduced functional tests for hate speech detection models. However, these tests currently only exist for English-language content, which means that they cannot support the development of more effective models in other languages spoken by billions across the world. To help address this issue, we introduce Multilingual HateCheck (MHC), a suite of functional tests for multilingual hate speech detection models. MHC covers 34 functionalities across ten languages, which is more languages than any other hate speech dataset. To illustrate MHC{'}s utility, we train and test a high-performing multilingual hate speech detection model, and reveal critical model weaknesses for monolingual and cross-lingual applications.",
+        }
+        """,
+        n_samples={"test": 1907},
+        avg_character_length={"test": 49.7},
+    )
+
+    def dataset_transform(self):
+        keep_cols = ["test_case", "label_gold"]
+        rename_dict = dict(zip(keep_cols, ["text", "label"]))
+        remove_cols = [
+            col for col in self.dataset["test"].column_names if col not in keep_cols
+        ]
+        self.dataset = self.dataset.rename_columns(rename_dict)
+        self.dataset = self.dataset.class_encode_column("label")
+        self.dataset = self.dataset.class_encode_column("functionality")
+        self.dataset = self.dataset["test"].train_test_split(
+            test_size=0.5, seed=42, stratify_by_column="functionality"
+        )  # balanced sampling across types of hate speech
```

Comment:
2 things here:

- I think you should stratify on the labels since that's what the classifier will use, and your dataset is imbalanced: `stratify_by_column="label"`. Altough, I find it interesting to stratify on "functionality", I think that if you'd liek to do taht with sklearn, you'll have to concatenate columns "label" and this one and stratify on it.
- You can use: `self.seed` for the seed value.

Reply:
Regarding the last point, this is how the split looks like (alreday stratified by labels). Is that alright?
```
DatasetDict({
    train: Dataset({
        num_rows: 1907
    })
    test: Dataset({
        num_rows: 1908
    })
})
```